test/e2e/deployment.go

```diff
@@ -270,3 +273,127 @@ func testRollingUpdateDeploymentEvents(f *Framework) {
 	Expect(events.Items[0].Message).Should(Equal(fmt.Sprintf("Scaled up rc %s to 1", newRC.Name)))
 	Expect(events.Items[1].Message).Should(Equal(fmt.Sprintf("Scaled down rc %s to 0", rcName)))
 }
+
+func testRolloverDeployment(f *Framework) {
+	ns := f.Namespace.Name
+	c := f.Client
+	deploymentPodLabels := map[string]string{"name": "sample-pod"}
+	rcPodLabels := map[string]string{
+		"name": "sample-pod",
+		"pod":  "redis",
+	}
+	rcPodLabels2 := map[string]string{
+		"name": "sample-pod",
+		"pod":  "redis-slave",
+	}
+
+	rcName := "redis-controller"
+	_, err := c.ReplicationControllers(ns).Create(&api.ReplicationController{
+		ObjectMeta: api.ObjectMeta{
+			Name:   rcName,
+			Labels: deploymentPodLabels,
+		},
+		Spec: api.ReplicationControllerSpec{
+			Replicas: 2,
+			Selector: rcPodLabels,
+			Template: &api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: rcPodLabels,
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  "redis",
+							Image: "redis",
+						},
+					},
+				},
+			},
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		Logf("deleting replication controller %s", rcName)
+		Expect(c.ReplicationControllers(ns).Delete(rcName)).NotTo(HaveOccurred())
+	}()
+	rcName2 := "redis-slave-controller"
+	_, err = c.ReplicationControllers(ns).Create(&api.ReplicationController{
```

Comment:
Instead of 2 existing RCs, a better test will be when the second RC was created by this deployment (deployment should start scaling down the RC that it was scaling up).
1. Create an RC and then create a deployment.
2 Verify that the deployment scales down the existing rc and starts scaling up a new RC.
3. Update the deployment, while it was still scaling up/down.
4. Verify that it is now scaling down both existing RCs and is scaling up a new RC.